pkg/client/observatorium/observability_config.go

```diff
@@ -252,29 +284,10 @@ func (c *ObservabilityConfiguration) ReadFiles() error {
 }
 
 func (c *ObservabilityConfiguration) Validate(env *environments.Env) error {
-	return c.ObservabilityCloudWatchLoggingConfig.validate()
-}
-
-func (c *ObservabilityConfiguration) ReadObservatoriumConfigFiles() error {
-	logsClientIdErr := shared.ReadFileValueString(c.LogsClientIdFile, &c.LogsClientId)
-	if logsClientIdErr != nil {
-		return logsClientIdErr
-	}
-
-	logsSecretErr := shared.ReadFileValueString(c.LogsSecretFile, &c.LogsSecret)
-	if logsSecretErr != nil {
-		return logsSecretErr
-	}
-
-	metricsClientIdErr := shared.ReadFileValueString(c.MetricsClientIdFile, &c.MetricsClientId)
-	if metricsClientIdErr != nil {
-		return metricsClientIdErr
-	}
-
-	metricsSecretErr := shared.ReadFileValueString(c.MetricsSecretFile, &c.MetricsSecret)
-	if metricsSecretErr != nil {
-		return metricsSecretErr
+	err := c.ObservabilityCloudWatchLoggingConfig.validate()
+	if err != nil {
+		return err
 	}
 
-	return nil
+	return c.DataPlaneObservabilityConfig.validate()
```

Comment:
I always prefer to do an explicit if err != nil check because although now that is the last thing that is validated in the future that might not be the case which will make you move this away from the return.
That is a personal preference though so I am good with either way.